Crashing Terraform

Hi, Have been using this fork in a docker container without troubles for the last couple weeks, today i rebuilt the docker to add some more tools and now Terraform Crashes on Plan

I assume it is related to the recent commit https://github.com/sl1pm4t/terraform-provider-kubernetes/commit/79e278cdca1ccd87cb65c6eea6da42e8734c3321#diff-35037ca50a08379030e056862dc341c5

```
panic: runtime error: invalid memory address or nil pointer dereference
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x136a32c]
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: 
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: goroutine 105 [running]:
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: github.com/sl1pm4t/terraform-provider-kubernetes/kubernetes.flattenDNSConfig(0x0, 0xc000aa7020, 0x16b4e04)
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: 	/go/src/github.com/sl1pm4t/terraform-provider-kubernetes/kubernetes/structures_pod.go:94 +0x3c
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: github.com/sl1pm4t/terraform-provider-kubernetes/kubernetes.flattenPodSpec(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc000245400, 0x1, 0x4, 0xc000aed130, ...)
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: 	/go/src/github.com/sl1pm4t/terraform-provider-kubernetes/kubernetes/structures_pod.go:25 +0x14a
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: github.com/sl1pm4t/terraform-provider-kubernetes/kubernetes.flattenPodTemplateSpec(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: 	/go/src/github.com/sl1pm4t/terraform-provider-kubernetes/kubernetes/structures_pod.go:113 +0x177
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: github.com/sl1pm4t/terraform-provider-kubernetes/kubernetes.flattenJobSpec(0xc000aed06c, 0xc000aed070, 0x0, 0xc000aed074, 0xc000959a20, 0xc000aed078, 0x0, 0x0, 0x0, 0x0, ...)
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: 	/go/src/github.com/sl1pm4t/terraform-provider-kubernetes/kubernetes/structure_job.go:46 +0x164
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: github.com/sl1pm4t/terraform-provider-kubernetes/kubernetes.resourceKubernetesJobRead(0xc000a9ed90, 0x156f740, 0xc000a26b70, 0xc000a9ed90, 0x0)
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: 	/go/src/github.com/sl1pm4t/terraform-provider-kubernetes/kubernetes/resource_kubernetes_job.go:173 +0x615
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: github.com/sl1pm4t/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc0005ea620, 0xc000962910, 0x156f740, 0xc000a26b70, 0xc000875800, 0x4c2801, 0x1417bc0)
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: 	/go/src/github.com/sl1pm4t/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:354 +0x160
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: github.com/sl1pm4t/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc0008a2850, 0xc0009628c0, 0xc000962910, 0xc000482000, 0x0, 0x7f535252ed80)
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: 	/go/src/github.com/sl1pm4t/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:308 +0x92
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: github.com/sl1pm4t/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc00097f920, 0xc00042ca70, 0xc00042cb50, 0x0, 0x0)
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: 	/go/src/github.com/sl1pm4t/terraform-provider-kubernetes/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:549 +0x4e
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: reflect.Value.call(0xc00008b020, 0xc0008e8b30, 0x13, 0x16aeb43, 0x4, 0xc0009aaf18, 0x3, 0x3, 0xc00017df40, 0xc0000180d0, ...)
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: 	/usr/local/go/src/reflect/value.go:447 +0x449
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: reflect.Value.Call(0xc00008b020, 0xc0008e8b30, 0x13, 0xc000494f18, 0x3, 0x3, 0x788e20, 0xc0000480d8, 0x0)
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: 	/usr/local/go/src/reflect/value.go:308 +0xa4
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: net/rpc.(*service).call(0xc000920d80, 0xc0009621e0, 0xc000503cb0, 0xc000503cc0, 0xc0004c6a80, 0xc0008edda0, 0x1417b80, 0xc00042ca70, 0x16, 0x1417bc0, ...)
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: 	/usr/local/go/src/net/rpc/server.go:384 +0x14e
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: created by net/rpc.(*Server).ServeCodec
2018-10-10T13:57:18.820Z [DEBUG] plugin.terraform-provider-kubernetes: 	/usr/local/go/src/net/rpc/server.go:481 +0x47e
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalRefresh, err: kubernetes_cluster_role.clusterdns_cluster_role: unexpected EOF
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalSequence, err: kubernetes_cluster_role.clusterdns_cluster_role: unexpected EOF
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalRefresh, err: kubernetes_job.kube-hunter: unexpected EOF
2018/10/10 13:57:18 [TRACE] [walkRefresh] Exiting eval tree: kubernetes_cluster_role.clusterdns_cluster_role
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalRefresh, err: kubernetes_config_map.clusterdns_config_map: unexpected EOF
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalRefresh, err: kubernetes_service.nginx_ingress_controller_stats: unexpected EOF
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalSequence, err: kubernetes_service.nginx_ingress_controller_stats: unexpected EOF
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalRefresh, err: kubernetes_service.nginx_ingress_controller: unexpected EOF
2018/10/10 13:57:18 [TRACE] root: eval: *terraform.EvalSequence
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalSequence, err: kubernetes_service.nginx_ingress_controller: unexpected EOF
2018/10/10 13:57:18 [TRACE] root: eval: *terraform.EvalGetProvider
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalRefresh, err: kubernetes_service.nginx_ingress_backend: unexpected EOF
2018/10/10 13:57:18 [TRACE] [walkRefresh] Exiting eval tree: kubernetes_service.nginx_ingress_controller
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalSequence, err: kubernetes_service.nginx_ingress_backend: unexpected EOF
2018/10/10 13:57:18 [TRACE] root: eval: *terraform.EvalReadState
2018/10/10 13:57:18 [TRACE] [walkRefresh] Exiting eval tree: kubernetes_service.nginx_ingress_backend
2018/10/10 13:57:18 [TRACE] [walkRefresh] Exiting eval tree: kubernetes_service.nginx_ingress_controller_stats
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalSequence, err: kubernetes_job.kube-hunter: unexpected EOF
2018/10/10 13:57:18 [TRACE] root: eval: *terraform.EvalRefresh
2018/10/10 13:57:18 [TRACE] [walkRefresh] Exiting eval tree: kubernetes_job.kube-hunter
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalRefresh, err: kubernetes_service_account.nginx_ingress_controller: unexpected EOF
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalSequence, err: kubernetes_service_account.nginx_ingress_controller: unexpected EOF
2018/10/10 13:57:18 [TRACE] [walkRefresh] Exiting eval tree: kubernetes_service_account.nginx_ingress_controller
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalRefresh, err: kubernetes_service_account.clusterdns_service_account: unexpected EOF
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalSequence, err: kubernetes_service_account.clusterdns_service_account: unexpected EOF
2018/10/10 13:57:18 [TRACE] [walkRefresh] Exiting eval tree: kubernetes_service_account.clusterdns_service_account
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalRefresh, err: kubernetes_deployment.azure-oauth2-api-proxy-deployment: connection is shut down
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalSequence, err: kubernetes_deployment.azure-oauth2-api-proxy-deployment: connection is shut down
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalRefresh, err: kubernetes_service.nginx_ingress_controller_metrics: unexpected EOF
2018/10/10 13:57:18 [TRACE] [walkRefresh] Exiting eval tree: kubernetes_deployment.azure-oauth2-api-proxy-deployment
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalSequence, err: kubernetes_service.nginx_ingress_controller_metrics: unexpected EOF
2018/10/10 13:57:18 [TRACE] [walkRefresh] Exiting eval tree: kubernetes_service.nginx_ingress_controller_metrics
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalSequence, err: kubernetes_config_map.clusterdns_config_map: unexpected EOF
2018/10/10 13:57:18 [TRACE] [walkRefresh] Exiting eval tree: kubernetes_config_map.clusterdns_config_map
2018-10-10T13:57:18.823Z [DEBUG] plugin: plugin process exited: path=/root/.terraform.d/plugins/terraform-provider-kubernetes
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "kubernetes_cluster_role_binding.nginx_ingress_controller"
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalRefresh, err: kubernetes_config_map.nginx_ingress_controller: unexpected EOF
2018/10/10 13:57:18 [ERROR] root: eval: *terraform.EvalSequence, err: kubernetes_config_map.nginx_ingress_controller: unexpected EOF
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "kubernetes_deployment.clusterdns_deployment"
2018/10/10 13:57:18 [TRACE] [walkRefresh] Exiting eval tree: kubernetes_config_map.nginx_ingress_controller
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "kubernetes_deployment.nginx_ingress_deployment"
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "null_resource.kubernetes_certmanager_crds"
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "provider.null (close)"
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "kubernetes_role_binding.nginx_ingress_role_binding"
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "kubernetes_cluster_role.cert_manager_view"
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "kubernetes_cluster_role.cert_manager_edit"
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "kubernetes_namespace.cert-manager"
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "kubernetes_service_account.cert_manager"
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "kubernetes_cluster_role.cert-manager"
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "kubernetes_deployment.cert-manager"
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "kubernetes_cluster_role_binding.cert_manager"
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "provider.kubernetes (close)"
2018/10/10 13:57:18 [TRACE] dag/walk: upstream errored, not walking "root"
2018/10/10 13:57:18 [DEBUG] plugin: waiting for all plugin processes to complete...
2018-10-10T13:57:18.957Z [WARN ] plugin: error closing client during Kill: err="connection is shut down"
2018-10-10T13:57:18.957Z [DEBUG] plugin.terraform-provider-tls_v1.2.0_x4: 2018/10/10 13:57:18 [ERR] plugin: plugin server: accept unix /tmp/plugin394915303: use of closed network connection
2018-10-10T13:57:18.957Z [DEBUG] plugin.terraform-provider-azurerm_v1.16.0_x4: 2018/10/10 13:57:18 [ERR] plugin: plugin server: accept unix /tmp/plugin804329139: use of closed network connection
2018-10-10T13:57:18.959Z [DEBUG] plugin.terraform: local-exec-provisioner (internal) 2018/10/10 13:57:18 [ERR] plugin: plugin server: accept unix /tmp/plugin663217653: use of closed network connection
2018-10-10T13:57:18.959Z [DEBUG] plugin.terraform: local-exec-provisioner (internal) 2018/10/10 13:57:18 [DEBUG] plugin: waiting for all plugin processes to complete...
2018-10-10T13:57:18.959Z [DEBUG] plugin: plugin process exited: path=/builds/tf-k8s-kafka-demo/.terraform/plugins/linux_amd64/terraform-provider-tls_v1.2.0_x4
2018-10-10T13:57:18.960Z [DEBUG] plugin: plugin process exited: path=/go/bin/terraform
2018-10-10T13:57:18.961Z [DEBUG] plugin: plugin process exited: path=/builds/tf-k8s-kafka-demo/.terraform/plugins/linux_amd64/terraform-provider-null_v1.0.0_x4
2018-10-10T13:57:18.963Z [DEBUG] plugin: plugin process exited: path=/builds/tf-k8s-kafka-demo/.terraform/plugins/linux_amd64/terraform-provider-azurerm_v1.16.0_x4



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.
```

--- a/kubernetes/structures_pod.go
+++ b/kubernetes/structures_pod.go
@@ -89,19 +89,20 @@ func flattenPodSpec(in v1.PodSpec) ([]interface{}, error) {
 }
 
 func flattenDNSConfig(in *v1.PodDNSConfig) interface{} {
-	att := make(map[string]interface{})
+	if in != nil {
+		att := make(map[string]interface{})
+		att["nameservers"] = in.Nameservers
 
-	att["nameservers"] = in.Nameservers
+		optMap := make(map[string]string, len(in.Options))
+		for _, opt := range in.Options {
+			optMap[opt.Name] = *opt.Value
+		}
+		att["options"] = optMap
 
-	optMap := make(map[string]string, len(in.Options))
-	for _, opt := range in.Options {
-		optMap[opt.Name] = *opt.Value
+		att["searches"] = in.Searches
+		return []interface{}{att}
 	}
-	att["options"] = optMap
-
-	att["searches"] = in.Searches
-
-	return []interface{}{att}
+	return nil
 }
 
 func flattenPodTemplateSpec(in v1.PodTemplateSpec, d *schema.ResourceData) ([]interface{}, error) {
@@ -526,7 +527,7 @@ func expandPodSpec(p []interface{}) (v1.PodSpec, error) {
 
 func expandDNSConfig(l []interface{}) *v1.PodDNSConfig {
 	if len(l) == 0 || l[0] == nil {
-		return &v1.PodDNSConfig{}
+		return nil
 	}
 
 	in := l[0].(map[string]interface{})

